### PR TITLE
[DDW-1213] Optimize E2E tests

### PR DESCRIPTION
**Pull request by daniloprates**
This PR optimizes E2E tests, standardizing some methods and removing unnecessary code.

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1583784166187600)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@DominikGuzei @tomothespian please review this one today 🙏

**Comment by nikolaglumac** [comment]
@daniloprates I still get this failure :(

Screenshot 2020-03-10 at 10 29 51

**Comment by tomothespian** [comment]
@daniloprates I got same failure as Nikola :(

**Comment by nikolaglumac** [comment]
@tomothespian can you please check this? We need to fix it and merge this PR ASAP 🙏

**Comment by daniloprates** [comment]
> @daniloprates I still get this failure :(
> 

@nikolaglumac @tomothespian fixed. The issue was happening when running this scenario for the second time. E.G. if it failed due to timeout (a cardano-sl issue), on the second run the `balance wallet with funds` wouldn't have funds, anymore. Therefore now, in case the imported balance wallet has no funds, it will throw an error, asking to reset the Daedalus directory.

**Comment by tomothespian** [comment]
@daniloprates I'm still getting random test failures. Is that also case on your side?

**Comment by daniloprates** [comment]
> @daniloprates I'm still getting random test failures. Is that also case on your side?

Yes. Mostly timeout ones. But there's one that breaks cardano node. Investigating... ⏳

**Comment by tomothespian** [comment]
@daniloprates yeah same on mi side, almost 90% failed because timeout.
Thanks Danilo ❤️

**Comment by nikolaglumac** [comment]
@daniloprates I still get the same error like before :(

**Comment by daniloprates** [comment]
> @daniloprates I still get the same error like before :(
> 

@nikolaglumac yes, I didn't add any new fix for that yet. I'm off this morning but will make sure to discover some magic for this and the other issues by the EOD.

**Comment by daniloprates** [comment]
I have added a fix for the aforementioned issue. The `this.wallets` array was not updating the wallet amount correctly. Not it gets the wallet amount directly from the client, in a string format, since webdriver doesn't import BigNumber properly.

**Comment by nikolaglumac** [comment]
@daniloprates it seems there are new problems now :(
Screenshot 2020-03-15 at 20 05 54

**Comment by daniloprates** [comment]
@nikolaglumac random timeout issues still happen and there's nothing we can do ATM

**Comment by daniloprates** [comment]
@nikolaglumac we've run there tests numerous times and have always faced at least one timeout issue, always different from the previous.

**Comment by nikolaglumac** [comment]
@daniloprates OK. Let's discuss this when you start today 🙏

**Comment by tomothespian** [comment]
@daniloprates I think that I found an issue.
Soo problem is with event emitter which breaks IPC and caused Cardano Node crashes. Event emitter comes from `events.js` and that is from `NodeJS`. I tried to update NodeJs version and ran tests 8 times. I didn't get Cardano Node crash, unexpected timeouts... Tests pass multiple times and just 2 times failed but the error is reasonable (Transfer funds scenario - transfer took a little bit longer than default timeout and another Scenario: `Balance wallet "Create a Rewards wallet" notification is displayed if the wallet is NOT empty and I don't have a Rewards wallet in the UI` which we need to check).
So my suggestion is to wait for https://github.com/input-output-hk/daedalus/pull/1883 to be merged . Maybe also to wait https://github.com/input-output-hk/daedalus/pull/1887 so we can update `spectron` inside this optimization PR. What you think?

cc @nikolaglumac

**Comment by daniloprates** [comment]
> @daniloprates I think that I found an issue.

@tomothespian sounds good!

File: package.json [hunk 1]
```diff
@@ -156,6 +161,7 @@
     "chroma-js": "2.1.0",
     "classnames": "2.2.6",
     "csv-stringify": "5.3.6",
+    "cucumber-html-reporter": "^5.1.0",
```
Comment (nikolaglumac) [inline package.json]: @daniloprates @tomothespian please use fixed version here!

**Pull request merged**